main.py

```diff
@@ -107,7 +110,7 @@ async def health_check():
 
 
 @app.get("/documents", response_model=list[DocumentResponse])
-async def get_documents_by_ids(ids: list[str]):
+async def get_documents_by_ids(ids: list[str] = Query(...)):
```

Comment:
This is the key of my change.  Without this, I am getting a 422 error of FastAPI. Note that I am using python-3.12 which might be the difference to the originally tested python-3.10.